### ISSUE TITLE
After reading NeumannIC, I still couldn't solve the wave equation

### ISSUE THREAD

**Issue by 979736316** [lululxvi/deepxde#146]
I want to use Deepxde to solve the wave equation.
![image](https://user-images.githubusercontent.com/52651366/96898540-6a9fca80-14c2-11eb-8e1f-20710b1d7ef5.png)
In ut (0, x) = Π sin（3Πx） processing, I refer to the link "[(https://github.com/lululxvi/deepxde/issues/127)]", specific code is as follows.I don't know what I did wrong
```
`from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import numpy as np

import deepxde as dde
from deepxde.backend import tf
import matplotlib.pyplot as plt
import pandas as pd

def main():
    def pde(x, u):
        du_x = tf.gradients(u, x)[0]
        du_x, du_t = du_x[:, 0:1], du_x[:, 1:2]
        du_xx = tf.gradients(du_x, x)[0][:, 0:1]
        du_tt = tf.gradients(du_t, x)[0][:, 1:2]
        return du_tt - du_xx

    def func(x):
        u = 0.5*np.sin(np.pi*x[:,0:1])*np.cos(np.pi*x[:,1:2])+(1/3)*np.sin(3*np.pi*x[:,0:1])*np.sin(3*np.pi*x[:,1:2])
        return u

    geom = dde.geometry.Interval(0, 1)
    timedomain = dde.geometry.TimeDomain(0, 1)
    geomtime = dde.geometry.GeometryXTime(geom, timedomain)

    def fun_initbc_all_a(x):
        return 0.5*np.sin(np.pi*x[:,0:1])
    def dt_fun_init_all_b(x):
        return np.pi*np.sin(3*np.pi*x[:,0:1])
    # def dt_fun_init_all_c(x):
    #     return np.zeros((len(x), 1))

    def boundary_initial(x, _):
        return np.isclose(x[-1], 0)

    ic = dde.IC(geomtime, fun_initbc_all_a, lambda _, on_initial: on_initial)
    bc_init = dde.NeumannBC(geomtime, dt_fun_init_all_b, boundary_initial)
    bc_a = dde.DirichletBC(
        geomtime, lambda x: np.zeros((len(x), 1)), lambda _, on_boundary: on_boundary
    )

    data = dde.data.TimePDE(
        geomtime,
        pde,
        [bc_init, bc_a, ic],
        num_domain=2450,
        num_boundary=160,
        num_initial=80,
        solution=func,
        num_test=50000,
    )

    net = dde.maps.FNN([2] + [50] * 3 + [1], "tanh", "Glorot normal")
    # net.apply_output_transform(lambda x, y: y * 10)
    model = dde.Model(data, net)

    model.compile("adam", lr=1e-3, metrics=["l2 relative error"])
    losshistory, train_state = model.train(epochs=5000, display_every=100)
    model.compile("L-BFGS-B", metrics=["l2 relative error"])
    # model.compile("L-BFGS-B",metrics=["l2 relative error"],loss_weights = [1,100,100,100])
    losshistory, train_state = model.train()
    dde.saveplot(losshistory, train_state, issave=True, isplot=True)
    x = np.linspace(0, 1, 200)
    t = np.linspace(0, 1, 200)
    X = np.vstack((x, t)).T
    y_true = func(X)
    y_pred = model.predict(X)
    f = model.predict(X, operator=pde)
    print("Mean residual:", np.mean(np.absolute(f)))
    print("L2 relative error:", dde.metrics.l2_relative_error(y_true, y_pred))
    np.savetxt("test.dat", np.hstack((X, y_true, y_pred)))

if __name__ == "__main__":
    main()
`
```

**Comment by 979736316** [comment]
![image](https://user-images.githubusercontent.com/52651366/96900616-e3078b00-14c4-11eb-84fe-895ad2addbd1.png)
It can be seen from the monitored loss value that there is an error in the definition of the second loss value

**Comment by zxw4688** [comment]
The derivative of initial value on time is not right. You should use the dde.OperatorBC to define it.

**Comment by 979736316** [comment]
> The derivative of initial value on time is not right. You should use the dde.OperatorBC to define it.
Thanks for your reply. Can you teach me the definition of this problem OperatorBC.

**Comment by 979736316** [comment]
> 时间的初始值的导数不正确。您应该使用dde.OperatorBC进行定义。

Thank you again for your reply. I have solved this problem

**Comment by katayooneshkofti** [comment]
@979736316 is it possible for you to upload the final code?
